eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/Imp/RatifySpec.hs

```diff
@@ -0,0 +1,166 @@
+{-# LANGUAGE NumericUnderscores #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Test.Cardano.Ledger.Conway.Imp.RatifySpec (spec) where
+
+import Cardano.Ledger.BaseTypes
+import Cardano.Ledger.Coin
+import Cardano.Ledger.Conway.Core
+import Cardano.Ledger.Conway.Governance
+import Data.Default.Class (def)
+import qualified Data.Map.Strict as Map
+import Lens.Micro
+import Test.Cardano.Ledger.Conway.ImpTest
+import Test.Cardano.Ledger.Imp.Common
+
+spec ::
+  forall era.
+  ConwayEraImp era =>
+  SpecWith (ImpTestState era)
+spec =
+  describe "RATIFY" $ do
+    delayingActionsSpec
+
+delayingActionsSpec ::
+  forall era.
+  ConwayEraImp era =>
+  SpecWith (ImpTestState era)
+delayingActionsSpec =
+  describe "Delaying actions" $ do
+    it "A delaying action delays its child even when both ere proposed and ratified in the same epoch" $ do
+      (dRep, committeeMember, _gpi) <- electBasicCommittee
+      conwayModifyPParams $ ppGovActionLifetimeL .~ EpochInterval 5
+      gai0 <- submitInitConstitutionGovAction
+      gai1 <- submitChildConstitutionGovAction gai0
+      gai2 <- submitChildConstitutionGovAction gai1
+      gai3 <- submitChildConstitutionGovAction gai2
+      submitYesVote_ (DRepVoter dRep) gai0
+      submitYesVote_ (CommitteeVoter committeeMember) gai0
+      submitYesVote_ (DRepVoter dRep) gai1
+      submitYesVote_ (CommitteeVoter committeeMember) gai1
+      submitYesVote_ (DRepVoter dRep) gai2
+      submitYesVote_ (CommitteeVoter committeeMember) gai2
+      submitYesVote_ (DRepVoter dRep) gai3
+      submitYesVote_ (CommitteeVoter committeeMember) gai3
+      passNEpochs 2
+      getLastEnactedConstitution `shouldReturn` SJust (GovPurposeId gai0)
+      passEpoch
+      getLastEnactedConstitution `shouldReturn` SJust (GovPurposeId gai1)
+      passEpoch
+      getLastEnactedConstitution `shouldReturn` SJust (GovPurposeId gai2)
+      passEpoch
+      getLastEnactedConstitution `shouldReturn` SJust (GovPurposeId gai3)
+      getConstitutionProposals `shouldReturn` Map.empty
+    it "A delaying action delays all other actions even when all of them may be ratified in the same epoch" $ do
+      (dRep, committeeMember, _gpi) <- electBasicCommittee
+      void $ conwayModifyPParams $ ppGovActionLifetimeL .~ EpochInterval 5
```

Comment:
What's up with `void`? :grin: It returns `()`
```suggestion
      modifyPParams $ ppGovActionLifetimeL .~ EpochInterval 5
```

Reply:
Oh that's a left over :sweat: :pray: